core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,157 @@
+
+        Returns:
+            str. Description of invalid tags property.
+        """
+        output_string = ''
+        visited = set()
+        dup: List[str] = []
+        empty_tag = max_length_exceed_tag = 0
+
+        if len(tags) > 10:
+            output_string += ' tags length more than 10,'
+        if len(set(tags)) < len(tags):
+            for ele in tags:
+                if ele.strip() != '':
+                    if ele in visited or (visited.add(ele)):
+                        dup.append(ele)
+            if len(dup) != 0:
+                output_string += f' {len(dup)} duplicate values {dup},'
+
+        for tag in tags:
+            if tag.strip() == '':
+                empty_tag += 1
+            elif len(tag) > 30:
+                max_length_exceed_tag += 1
+
+        if (empty_tag != 0) or (max_length_exceed_tag != 0):
+            output_string += (
+                f' {empty_tag} empty tag and'
+                + f' {max_length_exceed_tag} tag having length more than 30, ')
+
+        last_comma_index = output_string.rfind(',')
+        output_string = (
+            output_string[:last_comma_index] + '.'
+            + output_string[last_comma_index + 1:])
+
+        return output_string.rstrip()
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of invalid explorations with their id and
+        actual tags list.
+
+        Returns:
+            PCollection. Returns PCollection of invalid explorations with
+            their id and actual tags list.
+        """
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+        )
+
+        exp_ids_with_invalid_tags = (
+            total_explorations
+            | 'Combine exploration tags and ids' >> beam.Map(
+                lambda exp: (exp.id, exp.tags))
+            | 'Filter exploraton with tags length greater than 10' >>
+                beam.Filter(lambda exp: self._check_tags_validity(exp[1]))
+        )
+
+        report_number_of_exps_queried = (
+            total_explorations
+            | 'Report count of exp models' >> (
+                job_result_transforms.CountObjectsToJobRunResult('EXPS'))
+        )
+
+        report_number_of_invalid_exps = (
+            exp_ids_with_invalid_tags
+            | 'Report count of invalid exp models' >> (
+                job_result_transforms.CountObjectsToJobRunResult('INVALID'))
+        )
+
+        report_invalid_ids_and_their_actual_len = (
+            exp_ids_with_invalid_tags
+            | 'Save info on invalid exps' >> beam.Map(
+                lambda objects: job_run_result.JobRunResult.as_stderr(
+                    'The exp of id %s contains%s'
```

Comment:
```suggestion
                    'The exp of id %s contains%s'
```
'The exp of id %s contains %s'

Reply:
Done!